openproblems/tasks/spatial_decomposition/datasets/_utils.py

```diff
@@ -0,0 +1,88 @@
+from .._utils import obs_means
+from anndata import AnnData
+
+import numpy as np
+import random
+
+
+def generate_seq(length=14):
+    return "".join(random.choice("CGTA") for _ in range(length))
+
+
+# pass the reference data
+def generate_synthatic_data(adata, sim_type="avg"):
+    n_genes = adata.n_var
+    n_cells = adata.n_obs
+    n_types = len(set(adata.obs["label"].values))
+
+    # TODO(make these arguments)
+    bead_depth = 1000
+    num_of_beads = n_cells * 2
+    # generate proportion values
+    props = np.random.dirichlet(np.ones(n_types), num_of_beads)
+
+    true_proportion = np.zeros((num_of_beads, n_types))
+    bead_to_gene_matrix = np.zeros((num_of_beads, n_genes))
+
+    # if sim_type avg
+    # generate from avg profiles
+    if sim_type == "avg":
+        profile_mean = obs_means(adata, "label")
```

Comment:
`profile_means` needs to be normalized to avoid unexpected behaviour in `np.random.multinomial` where it's used as probabilities. See [link](https://numpy.org/doc/stable/reference/random/generated/numpy.random.multinomial.html)